src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +36,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        String[] details = address.split(",", ADDRESS_DETAIL_COUNT);
+        this.block = new Block(details[ADDRESS_DETAIL_INDEX_BLOCK].trim());
+        this.street = new Street(details[ADDRESS_DETAIL_INDEX_STREET].trim());
+        this.unit =  new Unit(details[ADDRESS_DETAIL_INDEX_UNIT].trim());
+        this.postalCode =  new PostalCode(details[ADDRESS_DETAIL_INDEX_POSTAL_CODE].trim());
+        this.value = toString();
```

Comment:
Not acceptable to retain original address value for “convenience”; unless there is a requirement that the formatting is different in user specified string, no need to retain.